src/argilla/client/feedback/training/schemas.py

```diff
@@ -1602,7 +1602,14 @@ def _prepare_for_training_with_sentence_transformers(
             raise ValueError("The dataset must contain at least one sample to be able to train.")
 
         # Use the first sample to decide what type of dataset to generate:
-        sample_keys = set(data[0].keys())
+        if isinstance(data[0], list):
+            # In case we are returning lists, extract the first element of that list to check the fields.
+            sample_keys = set(data[0][0].keys())
+        elif isinstance(data[0], dict):
+            sample_keys = set(data[0].keys())
+        else:
+            raise ValueError(f"The type is not supported: {type(data[0])}.")
+
         if sample_keys == {"label", "sentence-1", "sentence-2"}:
```

Comment:
maybe we should review these check too in order to align with a `set`? also, there seems to be a faulty line in 1618 checking for ` sample_keys == sample_keys`

Reply:
> maybe we should review these check too in order to align with a `set`? also, there seems to be a faulty line in 1618 checking for ` sample_keys == sample_keys`

I didn't noticed the error in line 1618.. thanks